pkg/execution/state/redis_state/lua/consumePause.lua

```diff
@@ -4,66 +4,26 @@ Consumes a pause.
 
 Output:
   0: Successfully consumed
-  1: Pause not found
-
+  1: Pause already consumed
 ]]
 
--- The pause ID is always provided as a key, as is the lease ID.
-local pauseKey      = KEYS[1]
-local pauseStepKey  = KEYS[2]
-local pauseEventKey = KEYS[3]
-local pauseInvokeKey = KEYS[4]
-local actionKey     = KEYS[5]
-local stackKey      = KEYS[6]
-local keyMetadata   = KEYS[7]
-local keyPauseAddIdx = KEYS[8]
-local keyPauseExpIdx = KEYS[9]
-local keyRunPauses   = KEYS[10]
-
-local pauseID      = ARGV[1]
-local invokeCorrelationId = ARGV[2]
-local pauseDataKey = ARGV[3] -- used to set data in run state store
-local pauseDataVal = ARGV[4] -- data to set
+local actionKey     = KEYS[1]
+local stackKey      = KEYS[2]
+local keyMetadata   = KEYS[3]
 
-local pause = redis.call("GET", pauseKey)
-if pause == false or pause == nil then
-	-- Pause no longer exists.
-	if pauseEventKey ~= "" then
-		-- Clean up regardless
-		redis.call("HDEL", pauseEventKey, pauseID)
-		-- Add an index of when the pause was added.
-		redis.call("ZREM", keyPauseAddIdx, pauseID)
-		-- Add an index of when the pause expires.  This lets us manually
-		-- garbage collect expired pauses from the HSET below.
-		redis.call("ZREM", keyPauseExpIdx, pauseID)
-	end
-	return 1
-end
-
-redis.call("DEL", pauseKey)
-redis.call("DEL", pauseStepKey)
--- SREM to remove the pause for this run
-redis.call("SREM", keyRunPauses, pauseID)
-
-if pauseEventKey ~= "" then
-	redis.call("HDEL", pauseEventKey, pauseID)
-end
+local pauseDataKey = ARGV[1] -- used to set data in run state store
+local pauseDataVal = ARGV[2] -- data to set
 
 if actionKey ~= nil and pauseDataKey ~= "" then
-	redis.call("RPUSH", stackKey, pauseDataKey)
-	redis.call("HSET", actionKey, pauseDataKey, pauseDataVal)
-	redis.call("HINCRBY", keyMetadata, "step_count", 1)
-	redis.call("HINCRBY", keyMetadata, "state_size", #pauseDataVal)
-end
-
-if invokeCorrelationId ~= false and invokeCorrelationId ~= "" and invokeCorrelationId ~= nil then
-	redis.call("HDEL", pauseInvokeKey, invokeCorrelationId)
+  -- idempotency check: only ever consume a pause once
```

Comment:
Updating function state is not idempotent, so we have to introduce a check to prevent running this logic twice for the same pauseDataKey.